linera-core/src/local_node.rs

```diff
@@ -662,42 +655,39 @@ where
         None
     }
 
-    pub async fn download_blob<A>(
-        mut validators: Vec<(ValidatorName, A)>,
-        chain_id: ChainId,
+    pub async fn try_download_blob<A>(
+        mut validators: Vec<A>,
         blob_id: BlobId,
-    ) -> Option<HashedBlob>
+    ) -> Result<Option<HashedBlob>, LocalNodeError>
     where
         A: LocalValidatorNode + Clone + 'static,
     {
         // Sequentially try each validator in random order.
         validators.shuffle(&mut rand::thread_rng());
-        for (name, mut node) in validators {
-            if let Some(blob) =
-                Self::try_download_blob_from(name, &mut node, chain_id, blob_id).await
-            {
-                return Some(blob);
+        for mut node in validators {
+            if let Some(blob) = Self::try_download_blob_from(&mut node, blob_id).await? {
```

Comment:
We shouldn't fail if one faulty/malicious validator sends us the wrong blob. We should continue trying to download it from someone else.